spyder_env_manager/spyder/widgets/main_widget.py

```diff
@@ -292,61 +352,304 @@ def on_close(self):
+            messages = ["Package", "Constraint", "Version"]
+            types = [
+                CustomParametersDialogWidgets.Label,
+                CustomParametersDialogWidgets.ComboBox,
+                CustomParametersDialogWidgets.LineEditVersion,
+            ]
+            contents = [
+                {package_info["name"]},
+                {"==", "<=", ">=", "<", ">", "latest"},
+                {},
+            ]
+            self._message_box_editable(
+                title,
+                messages,
+                contents,
+                types,
+                action=EnvironmentPackagesActions.InstallPackageVersion,
+                package_info=package_info,
+            )
+
     def _add_new_environment_entry(self, manager, action_result, result_message):
+        """
+        Handle the addition of a new Python envirionment in the GUI.
+
+        Add an entry in to the selected environment combobox and update config
+        for the selected environment.
+
+        Parameters
+        ----------
+        manager : envs_manager.manager.Manager
+            Python environment manager instance that is handling the environment.
+        action_result : bool
+            True if the environment creation was successful. False otherwise.
+        result_message : str
+            Message of the action result like message errors in case of issues.
+
+        Returns
+        -------
+        None.
+
+        """
         if action_result:
-            if self.envs == {"No environments available"}:
+            if not self.envs_available:
                 self.select_environment.clear()
             self.select_environment.addItem(manager.env_name, manager.env_directory)
             self.select_environment.setCurrentText(manager.env_name)
             self.set_conf("selected_environment", manager.env_name)
-            self.set_conf(
-                "environments_list", Manager.list_environments(CondaLikeInterface.ID)
+            self.envs_available = True
+        else:
+            self._message_error_box(result_message)
+        self.stop_spinner()
+
+    def _after_import_environment(self, manager, action_result, result_message):
+        """
+        Handle the creation of a new Python environment via the import functionality.
+
+        Parameters
+        ----------
+        manager : envs_manager.manager.Manager
+            Python environment manager instance that is handling the environment.
+        action_result : bool
+            True if the environment creation was successful. False otherwise.
+        result_message : str
+            Message of the action result like message errors in case of issues.
+
+        Returns
+        -------
+        None.
+
+        """
+        if action_result:
+            # Add new imported environment entry
+            self._add_new_environment_entry(manager, action_result, result_message)
+            # Install needed spyder-kernels version
```

Comment:
```suggestion
        # Add new imported environment entry
        self._add_new_environment_entry(manager, action_result, result_message)

        # Install needed spyder-kernels version
        if action_result:
```

* I suggest to do this change because `_add_new_environment_entry` already has the logic to show `result_message` in case of env creation failure.
* The part about installing Spyder-kernels should only be necessary if the environment has Python and it's not already present on it. But we could work out those details later.